components/_util/transKeys.ts

```diff
@@ -0,0 +1,25 @@
+export type mapItem = {
+  k: string;
+  i: number;
+};
+export type MapType = Record<string, mapItem>;
+
+export const groupKeysMap = (keys: string[]) => {
+  const map: MapType = {};
+  keys.forEach((k, i) => {
+    map[k] = { k, i };
```

Comment:
since `k` is already for the key. Just:  `map[key] = index`

PS: do not naming variable as `k` or `i`, which should be more readable like `key` or `index`

Reply:
the index value can be zero, which is a falsy value in javascript. should I use String(index)?

Reply:
Seems strange but ok. Pls add comment here to note why convert to string. Thx~